sdl-debezium/src/main/scala/io/smartdatalake/workflow/dataobject/DebeziumCdcDataObject.scala

```diff
@@ -0,0 +1,337 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2024 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.dataobject
+import com.typesafe.config.Config
+import io.debezium.embedded.Connect
+import io.debezium.engine.{ChangeEvent, DebeziumEngine}
+import io.smartdatalake.config.SdlConfigObject.{ConnectionId, DataObjectId}
+import io.smartdatalake.config.{FromConfigFactory, InstanceRegistry}
+import io.smartdatalake.debezium.{DebeziumChangeConsumer, DebeziumCompletionCallback, DebeziumSchemaConsumer, HasRecords}
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.workflow.ActionPipelineContext
+import io.smartdatalake.workflow.connection.DebeziumConnection
+import org.apache.kafka.connect.data.Schema.Type
+import org.apache.kafka.connect.data.{Field, Schema, Struct}
+import org.apache.kafka.connect.source.SourceRecord
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+import java.util
+import java.util.Properties
+import java.util.concurrent.{ExecutorService, Executors}
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+/**
+ * [[DataObject]] of type DebeziumCdcDataObject.
+ * Provides details to access Change data over Debezium Engine.
+ *
+ * @param id unique name of this data object
+ * @param connectionId optional id of [[io.smartdatalake.workflow.connection.DebeziumConnection]]
+ * @param table Source table to get change data from
+ * @param debeziumProperties Properties for the specific Debezium connector
+ * @param metadata optional data object metadata
```

Comment:
Can you add an example config in the doc? Something like this?
https://github.com/smart-data-lake/smart-data-lake/blob/master-spark3/sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/ColumnsTransformer.scala#L43